### DISCUSSION TITLE
Creating material properties of time and time step

### DISCUSSION THREAD

**Discussion by czadeh** [idaholab/moose#21339] [Q&A General] [answered]
Hi,

I wish to create a material property that has time dependent behavior. 
The function itself is dependent on cumulative time, but I am taking its derivative so that I can calculate the material property at each time step according to the derivative * delta t.

The form of the equation is as follows, where E is the material property I wish to calculate, i is the iteration, dE/dt is the time derivative, and delta_t is the time step:

E(i+1) = E(i) + dE(i)/dt * delta_t

My question is: how do I access

- delta_t
- cumulative t

within a standard material property header and source file? Is there some standard way of calling those values within the MOOSE framework? I am not executing with variable time step, but I want to avoid creating redundant variables and parameters.

Thanks,
Cameron

**Comment by GiudGiud** [comment, marked as answer]
Hello

Both the current time and the timestep are available in the Material Property code by accessing the FEProblem with the
`_mi_feproblem` class member. 
Then you get access to these FEProblemBase class routines:
```
  virtual Real & time() const { return _time; }
  virtual Real & timeOld() const { return _time_old; }
  virtual int & timeStep() const { return _t_step; }
  virtual Real & dt() const { return _dt; }
  virtual Real & dtOld() const { return _dt_old; }
```
by calling for example: `_mi_feproblem.dt()`

Guillaume

**Reply by czadeh** [reply]
Thank you!
